homeassistant/components/climate/daikin.py

```diff
@@ -0,0 +1,264 @@
+                )
+            ),
+            ATTR_SWING_MODE: list(
+                map(
+                    str.title,
+                    appliance.daikin_values(HA_ATTR_TO_DAIKIN[ATTR_SWING_MODE])
+                )
+            ),
+        }
+
+    def get(self, key):
+        """Retrieve device settings from API library cache."""
+        value = None
+        cast_to_float = False
+
+        if key in [ATTR_TEMPERATURE, ATTR_INSIDE_TEMPERATURE,
+                   ATTR_CURRENT_TEMPERATURE]:
+            value = self._api.device.values.get('htemp')
+            cast_to_float = True
+        if key == ATTR_TARGET_TEMPERATURE:
+            value = self._api.device.values.get('stemp')
+            cast_to_float = True
+        elif key == ATTR_OUTSIDE_TEMPERATURE:
+            value = self._api.device.values.get('otemp')
+            cast_to_float = True
+        elif key == ATTR_FAN_MODE:
+            value = self._api.device.represent('f_rate')[1].title()
+        elif key == ATTR_SWING_MODE:
+            value = self._api.device.represent('f_dir')[1].title()
+        elif key == ATTR_TARGET_TEMP_STEP:
+            return 1
+        elif key == ATTR_OPERATION_MODE:
+            import re
+
+            # Daikin can return also internal states auto-1 or auto-7
+            # and we need to translate them as AUTO
+            value = re.sub(
+                '[^a-z]',
+                '',
+                self._api.device.represent('mode')[1]
+            ).title()
+
+        if value is None:
+            _LOGGER.warning("Invalid value requested for key %s", key)
+        else:
+            if value == "-" or value == "--":
+                value = None
+            elif cast_to_float:
+                try:
+                    value = float(value)
+                except ValueError:
+                    value = None
+
+        return value
+
+    def set(self, settings):
+        """Set device settings using API."""
+        values = {}
+
+        for attr in [ATTR_TEMPERATURE, ATTR_FAN_MODE, ATTR_SWING_MODE,
+                     ATTR_OPERATION_MODE]:
+            if attr in settings and settings[attr] is not None:
+                value = settings[attr]
+                daikin_attr = HA_ATTR_TO_DAIKIN.get(attr)
+                if daikin_attr is not None:
+                    if value.title() in self._list[attr]:
+                        values[daikin_attr] = value.lower()
+                    else:
+                        _LOGGER.error("Invalid value %s for %s", attr, value)
+
+                # temperature
+                elif attr == ATTR_TEMPERATURE:
+                    try:
+                        values['stemp'] = str(int(value))
+                    except ValueError:
+                        _LOGGER.error("Invalid temperature %s", value)
+
+        if values:
+            self._api.device.set(values)
+            self._api.update(force_refresh=True)
```

Comment:
Since this is a polling entity (`should_poll` is set to True by default), the entity `update` method will be called at the end of all service calls, ie after all calls to this `set` method.

So you shouldn't call `self._api.update` from here, but instead just modify an instance attribute, eg `self.force_refresh`, to `True` and add a check of that instance attribute inside `update`, so that `update` will call `self._api.update(True)` instead of just `self._api.update()`. `update` should reset the instance attribute, `self.force_refresh`, to `False` after the call.